lib/src/rules/unnecessary_null_checks.dart

```diff
@@ -185,6 +186,16 @@ class _Visitor extends SimpleAstVisitor<void> {
   final LinterContext context;
   _Visitor(this.rule, this.context);
 
+  @override
+  void visitNullAssertPattern(NullAssertPattern node) {
+    if (node.operator.type != TokenType.BANG) return;
```

Comment:
There aren't any other operators that could appear there, so we don't need to test that it's the right one.